rosetta_generator/lib/src/tree/implementation/visitor.dart

```diff
@@ -138,9 +138,11 @@ class TranslationVisitor extends Visitor<TranslationProduct, TranslationNode> {
       MethodBuilder methodBuilder, Interceptor interceptor) {
     methodBuilder
       ..type = MethodType.getter
-      ..body = (node.isInHelper ? refThis : refInnerHelper)
-          .property(strResolveMethodName)
+      ..body = refResolve
           .call([keysClassRef.property(node.translation.keyVariable)]).code;
```

Comment:
This part fails in the generated subclasses:
![image](https://user-images.githubusercontent.com/26414044/82220550-1430cd80-991f-11ea-8445-a97fe931aabe.png)

Here we must call the Helper's resolve(...) method, like so:
![image](https://user-images.githubusercontent.com/26414044/82220843-7ab5eb80-991f-11ea-8a1b-a1c4641b19e9.png)

The "this" reference is only redundant in the "Future<void> load(Locale locale)" method, in the Helper.
This part should be updated for this to work:

...
 ..body = (node.isInHelper ? refThis : refInnerHelper)
...

The refThis should be replaced, or the ternary operator reworked .
Also, thank you for the PR! ^^

Reply:
I don't understand, where this class _$Good has come from and why the helper is injected there. The helper is an abstract class, not something you would call. How do I replicate this?

Reply:
oh I think I've got it.

Reply:
Put this in your json:
`{
  "good.luck": "Good luck!",
  "good.idea": "Good idea!"
}`

And this for you translation:
`part 'translation_grouped_dot_separator.g.dart';

@Stone(path: 'i18n', grouping: Grouping.withSeparator())
class TranslationGroupedDotSeparator
    with _$TranslationGroupedDotSeparatorHelper {
  static LocalizationsDelegate<TranslationGroupedDotSeparator> delegate =
      _$TranslationGroupedDotSeparatorDelegate();

  static TranslationGroupedDotSeparator of(BuildContext context) {
    return Localizations.of(context, TranslationGroupedDotSeparator);
  }

  @Intercept.simple()
  String simpleIntercept(String translation) => translation;
}`

Long version:
If you put the "good.idea" in the en.json, while a grouping withSeparator() is deffined, then you can reference it like this:
Translation.of(context).good.idea;

This is can be useful if you want to group strings, depending on the screen you are using it (or something).

For this to work Rosetta create the required sub-classes, which can be chained together, to make the above code work.

In the Example's case, it creates a "_$Good" class which hase a "idea" getter, which will return the resolved value, and it creates a _$Good"good" property in the Helper to refrerence it. It's kind of difficult do write down, but if you run the flutter_example you can see it in the generated files :D

Reply:
fixed, and tested with the underscore as well.

Reply:
Nice, thank you!